### ISSUE TITLE
Feature Request / Question re Roadmap

### ISSUE THREAD

**Issue by colingoldberg** [jensoleg/swagger-ui#21]
Hi,

This is not an issue as much as it is a question about where this tool is headed.

I have recently tried it out on a new swagger file and I love it! I have not configured some things, such as using Basic Auth (not OAuth2) and possibly allowing a 'default' api key to let people play, but I plan to do this soon.

My question is: If I incorporate this into my website, I would like to include content that is generally not included in a swagger file. eg. A general introduction to the API, things to look out for, examples of how to use the api, etc. Typically, this is catered for (in other tools, say) by allowing for markup.

In other words, I want the best of both worlds - swagger based plus markup based content.

Is there anything in the pipeline that may address this need in one tool?

Any comments?

Regards

Colin Goldberg

**Comment by jensoleg** [comment]
HI @colingoldberg 

The project is build upon the original swagger UI project. I have aimed to keep original functionality and only change the rendering of the definition file.

Adding extra content ideally should be backed by the generated definition files. Adding content to the generators will expand this project considerably and collide to much with the core development. 
Although we might come up with something as an inspiration to the swagger team ... maybe in an experimental fork.

I agree that extending the condensed Swagger UI information with usage information is necessary in many cases. You should be aware that you can deeplink to each endpoints in the UI from other documents. Also you can fork the project and suit it as you want.

To answer your question there is nothing in the pipeline to extend the project besides supporting the swagger schema. 

Ultimately this UI/theme or similar should be build in to the core swagger UI to avoid the projects drifting away.

**Comment by jensoleg** [comment]
I have received some requests on this issue and have come up with a solution (hack)  that may help you.

Follow  these steps :

Add pseudo endpoint to the swagger definition files like this:

> "/introduction/1": {
>   "head": {
>     "tags": [
>       "Introduction"
>     ],
>     "description": "Description #1 \n Nam sed condimentum est. Maecenas tempor sagittis sapien, nec rhoncus sem sagittis sit amet. Aenean at gravida augue, ac iaculis sem. Curabitur odio lorem, ornare eget elementum nec, cursus id lectus",
>     "summary": "Summary #1",
>     "operationId": "Intro#1"
>   }
> },
> "/introduction/2": {
>   "head": {
>     "tags": [
>       "Introduction"
>     ],
>     "description": "Description #2\n Nam sed condimentum est. Maecenas tempor sagittis sapien, nec rhoncus sem sagittis sit amet. Aenean at gravida augue, ac iaculis sem. Curabitur odio lorem, ornare eget elementum nec, cursus id lectus,
>     "summary": "Summary #2",
>     "operationId": “intro#2"
>   }
> },

Next you have to modify the js code a little bit to hide html stuff normally generated.
In the handlebar JS template file “operations.handlebar” 

`{{#ifneq parentId 'Introduction'}}
    <h3>
        <span class='http_method'>
            <a href='#!/{{encodedParentId}}/{{nickname}}'>{{method}}</a>
        </span>
        <span class='path'>
            <a href='#!/{{encodedParentId}}/{{nickname}}'
               class="{{#if deprecated}}deprecated{{/if}}">{{path}}</a>
        </span>
    </h3>
{{/ifneq}}`

and 

`{{#ifneq parentId 'Introduction'}}
    <form accept-charset="UTF-8" class="sandbox">
    </form>
{{/ifneq}}`

This will hide the controls.

You also have to define a new handlebar helper in the /helpers/handlebars.js file:

> Handlebars.registerHelper('ifneq', function (a, b, options) {
>     if (a != b) { return options.fn(this); }
> });

The result will display the endpoints with textual content only and you can customize further by adding properties to the api-explorer.css file  

![image](https://cloud.githubusercontent.com/assets/5655197/13378464/0921c34e-de08-11e5-9e7f-3635b9d57abf.png)
